sqldelight-compiler/src/test/kotlin/com/squareup/sqldelight/types/ValidatorTests.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.types
+
+import com.google.common.truth.Subject
+import com.google.common.truth.Truth
+import com.google.common.truth.Truth.assertThat
+import com.squareup.sqldelight.Status
+import com.squareup.sqldelight.util.parse
+import com.squareup.sqldelight.validation.QueryResults
+import com.squareup.sqldelight.validation.SqlDelightValidator
+import org.junit.Test
+import java.io.File
+
+class ValidatorTests {
+  private val testFile = File("src/test/data/ResolverTestData.sq")
+  private val parsed = parse(testFile)
+  private val symbolTable = SymbolTable() + SymbolTable(parsed, testFile)
+  private val status = SqlDelightValidator()
+      .validate(parsed, symbolTable) as Status.ValidationStatus.Validated
+
+  @Test
+  fun selectAll() {
+    assertThat(status.queries.withName("select_all"))
```

Comment:
I'm not a fan of how far away the queries are from these tests. Lots of cross-referencing has to happen. Can we get to a state (at some point) where they're embedded as strings inside their respective test?